### ISSUE TITLE
Very slow initialization of (re)connection to vpn

### ISSUE THREAD

**Issue by unnamed777** [AdguardTeam/AdGuardVPNForAndroid#292]
### Please answer the following questions for yourself before submitting an issue.

- [X] I am running the latest version
- [X] I checked the documentation and found no answer
- [X] I checked to make sure that this issue has not already been filed

### AdGuard VPN version

2.2.4 (52)

### Operating mode

VPN

### Environment

```markdown
- OS: Android 13 / Android 13
- Device: Samsung Tab S7 / Xiaomi 12S Ultra
- Firmware: One UI 5.1 / Xiaomi.eu 14.0.12.0
```

### Root access

- [ ] Yes, I have it.

### Issue Details

I've been experiecing an issue for the last month and a half. It could be split to two general cases.

Case 1. When I enable VPN via tile in notification shade, it tooks 20-30s to connect. In AdGuard VPN a list of servers is empty during this time. After >20 secs list of servers is showing up and vpn establishes connection.

Case 2. The connection is seems to be active (I see VPN icon in notification panel). But when it's used, speed is slow or nothing is loading at all. Usually I often have this problem when I use Instagram: new posts aren't loaded, in Stories neither pictures or videos are shown (or first 1-2 secs of first couple of stories are loaded and stuck after that). When I open AdGuard VPN, it shows status "Connected", but current server is marked as "Limited speed". Servers' list is empty. I have to wait 20-30s for loading of new servers' list. After that "Limited speed" disappears and everything works. The launch of the app is required, without it the servers' list won't update. All battery optimizations for the app are disabled.

The common of these cases is empty (obsolete?) list of available servers, and the issue disappears when the list is reloaded. It's reproduced on two devices (phone and tablet) and on two networks (wifi and lte).

I sent logs to devteam@adguard.com with subject "Logs for issue 292 AdGuard VPN for Android":
* adguard_vpn_logs_2.2.52_3004_134512 - 30.04 13:44, Samsung Tab S7. The app is opened via long tap on notification tile, it stucks for ~20s and reconnects then.
* adguard_vpn_logs_2.2.52_3004_204149 - 30.04 20:40, Samsung Tab S7. One of the stories is stuck, I opened the app, "limited speed" is showed, waited ~20s, servers' list is updated, everything starts working again.
* adguard_vpn_logs_2.2.52_3005_194654 - 30.05 19:45 (yep, a month later), Xiaomi 12S Ultra. If I'm not mistaken, I enabled vpn via notification shade, it was faded for 20-30s, and vpn connection was established after that.

**Comment by maxikuzmin** [comment]
@unnamed777 thanks for the long wait. Is the problem still recurring on version 2.3? 
We will look into your logs

**Comment by unnamed777** [comment]
> Is the problem still recurring on version 2.3?
Yes.
<details>
  <summary>Screenshots (the second taken with 20s delay)<summary>
  ![1](https://github.com/AdguardTeam/AdGuardVPNForAndroid/assets/1111833/49e7efe7-a483-4488-9df6-96950e90c301)
  ![2](https://github.com/AdguardTeam/AdGuardVPNForAndroid/assets/1111833/832fabbe-b17b-499f-a13b-fc58f516ae53)
</details>

**Comment by maxikuzmin** [comment]
@unnamed777 is this repeated on other mobile networks and other wifi? 
It seems like slow internet speeds

**Comment by unnamed777** [comment]
Yes, I've tested this on different wifi and mobile networks.

**Comment by maxikuzmin** [comment]
@unnamed777 please try updating to the latest version. If the problem is still recurring, then we will need your logs.

Here's what we need you to do:
- Open AdGuard VPN app settings;
- Advanced settings --> Logging level --> Extended;
- Reproduce the issue and remember the exact time it happened;
- Open Support --> Export logs and system info;
- Send the archive to devteam@adguard.com and mention this issue number in the subject.

**Comment by maxikuzmin** [comment]
@unnamed777 any news?

**Comment by unnamed777** [comment]
Case 2 is seems to be solved. I don't need to open AdGuard VPN to have working connection.
Case 1... Well, it's hard to tell. It's faster than 20s. How long should it take to establish vpn connection usually? I already forgot, how it was 6 month ago. Sometimes it takes 8 secs, somestimes 15 secs. When I kill AdGuard VPN intentionally (only for tests; all battery opmitizations and restrictions are disabled) and try to enable vpn via tile, it takes 5-10 secs. When I open the app (after killing), timing looks similar. I've tested on wifi and mobile networks.

**Comment by maxikuzmin** [comment]
@unnamed777 case 2 doesn't happen again for me. I hope the problem has been solved.
Case 1 - I repeated this. I noticed that after enabling AdGuard VPN using the tile in the notification bar, it does take a long time for AdGuard VPN to turn on at a glance, but if you pull the notification bar up/down, or just minimize it, the tile immediately shows up as "enabled". That is, we can see that AdGuard VPN is not enabled, although it is enabled, it's just that its icon is not highlighted.
It looks like a system bug, but we'll check it out
